WordPress/JetpackStatsWidgets/Remote service/StatsWidgetsService.swift

```diff
@@ -30,33 +32,21 @@ class StatsWidgetsService {
         }
         state = .loading
 
-        do {
-            let token = try SFHFKeychainUtils.getPasswordForUsername(AppConfiguration.Widget.Stats.keychainTokenKey,
-                                                                     andServiceName: AppConfiguration.Widget.Stats.keychainServiceName,
-                                                                     accessGroup: WPAppKeychainAccessGroup)
-            let wpApi = WordPressComRestApi(oAuthToken: token)
-            let service = StatsServiceRemoteV2(wordPressComRestApi: wpApi, siteID: widgetData.siteID, siteTimezone: widgetData.timeZone)
-
-            // handle fetching depending on concrete type
-            // we need to do like this as there is no unique service call
-            if let widgetData = widgetData as? HomeWidgetTodayData {
-                fetchTodayStats(service: service, widgetData: widgetData, completion: completion)
-            } else if let widgetData = widgetData as? HomeWidgetAllTimeData {
-                fetchAllTimeStats(service: service, widgetData: widgetData, completion: completion)
-            } else if let widgetData = widgetData as? HomeWidgetThisWeekData {
-                fetchThisWeekStats(service: service, widgetData: widgetData, completion: completion)
-            }
-        } catch {
-            completion(.failure(error))
-            self.state = .error
+        // handle fetching depending on concrete type
```

Comment:
I tested this and it works well, but I agree that a simpler fix might be better to reduce the chance of regressions this late in the beta. 

> It would probably be easier as a quick fix to retain the service on (existing) line 38 and set it to nil in completion.

I tested this in https://github.com/wordpress-mobile/WordPress-iOS/pull/new/fix/23087-stats-widget-widgets-do-not-update-automatically-review and it also works well.

Reply:
Yes, you're right. I wanted to do something clever but overcomplicated the fix in the process. Thanks for the review!